src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBEnumerableBuilder.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal class DocumentDBEnumerableBuilder<T> : IAsyncConverter<DocumentDBAttribute, IEnumerable<T>> where T : class
+    {
+        private readonly DocumentDBConfiguration _config;
+
+        public DocumentDBEnumerableBuilder(DocumentDBConfiguration config)
+        {
+            _config = config;
+        }
+
+        public async Task<IEnumerable<T>> ConvertAsync(DocumentDBAttribute attribute, CancellationToken cancellationToken)
+        {
+            DocumentDBContext context = _config.CreateContext(attribute);
+
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(context.ResolvedAttribute.DatabaseName, context.ResolvedAttribute.CollectionName);
+
+            List<T> finalResults = new List<T>();
+
+            string continuation = null;
+
+            SqlQuerySpec sqlSpec = new SqlQuerySpec
+            {
+                QueryText = context.ResolvedAttribute.SqlQuery,
+                Parameters = context.ResolvedAttribute.SqlQueryParameters ?? new SqlParameterCollection()
+            };
+
+            do
+            {
+                DocumentQueryResponse<T> response = await DocumentDBUtility.RetryAsync(
+                    () => context.Service.ExecuteNextAsync<T>(collectionUri, sqlSpec, continuation),
+                    context.MaxThrottleRetries);
```

Comment:
how does removing 'codesToIgnore' affect this? I'm assuming the not found will surface as a function error.. not sure you can get a not found for a query in any event.

One thing to consider is handling 429s - I know Parse had a ton of issues with too many requests errors causing strange bugs. Probably not in the scope of this initial codeflow, but may want to chat with the docdb team about it.